src/pages/saksbehandling/steg/beregning/FradragInputs.tsx

```diff
@@ -0,0 +1,290 @@
+                onChange={props.onChange}
+            />
+            <Normaltekst>{props.inputTekst}</Normaltekst>
+        </span>
+        <Feilmelding>{props.feil ?? ''}</Feilmelding>
+    </div>
+);
+
+const FradragsSelection = (props: {
+    label: string;
+    id: string;
+    className: string;
+    name: string;
+    value: string;
+    onChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
+    feil: string | undefined;
+    intl: IntlShape;
+    optionLabel: string;
+}) => (
+    <div>
+        <h3>{props.label}</h3>
+        <Select
+            onChange={props.onChange}
+            id={props.id}
+            name={props.name}
+            value={props.value}
+            className={props.className}
+        >
+            <option value="">{props.intl.formatMessage({ id: 'fradrag.type.emptyLabel' })}</option>
+            {Object.values(Fradragstype).map((f) => (
+                <option value={f} key={f}>
+                    {props.intl.formatMessage({ id: fradragstypeResourceId(f) })}
+                </option>
+            ))}
+        </Select>
+        <Feilmelding>{props.feil ? props.feil : ''}</Feilmelding>
+    </div>
+);
+
+const fraUtlandInntekt = yup
+    .object<FraUtlandInntektFormData>()
+    .defined()
+    .when('fraUtland', {
+        is: true,
+        then: yup.object<FraUtlandInntektFormData>({
+            beløpUtenlandskValuta: validateStringAsNumber,
+            valuta: yup.string().required(),
+            kurs: validateStringAsNumber,
+        }),
+        otherwise: yup.object<FraUtlandInntektFormData>(),
+    });
+
+const delerAvPeriode = yup
+    .object<DelerAvPeriodeFormData>()
+    .defined()
+    .when('delerAvPeriodeChecked', {
+        is: true,
+        then: yup.object<DelerAvPeriodeFormData>({
+            fraOgMed: yup.date().required().nullable(),
+            tilOgMed: yup.date().required().nullable(),
+        }),
+        otherwise: yup.object<DelerAvPeriodeFormData>(),
+    });
+
+export const fradragSchema = yup.object<FradragFormData>({
+    beløp: validateStringAsNumber,
+    type: yup.string().defined().oneOf(Object.values(Fradragstype), 'Du må velge en fradragstype'),
+    fraUtland: yup.boolean(),
+    fraUtlandInntekt: fraUtlandInntekt,
+    delerAvPeriodeChecked: yup.boolean(),
+    delerAvPeriode: delerAvPeriode,
+});
+
+export const isValidFradrag = (f: unknown): f is Fradrag => fradragSchema.isValidSync(f);
+
+export const FradragInputs = (props: {
+    fradrag: Array<FradragFormData>;
+    feltnavn: string;
+    errors: string | string[] | FormikErrors<FradragFormData>[] | undefined;
+    intl: IntlShape;
+    onChange: (e: React.ChangeEvent<unknown>) => void;
+    setFieldValue: (field: string, value: Date | [Date, Date] | null | string) => void;
+    onLeggTilClick: () => void;
+    onFjernClick: (index: number) => void;
+}) => {
+    return (
+        <div className={styles.fradragContainer}>
+            {typeof props.errors === 'string' && props.errors}
```

Comment:
Må man checka efter typen string här? Vad innebärer det att props.errors er en string istället för ett object?

Reply:
Tror Ingar er den beste å spørre det. Jeg har testet litt uten den, og alt ser ut til å funke fint, men usikker på hvor viktig den er. skal sjekke med han når jeg kan :)

Reply:
Her må man sjekke for `string` ja. Hvis `props.errors` er et objekt så vil man få runtimefeil fordi objekter ikke er gyldige React-barn.

Reply:
@ingara men varför kan den antingen vara object eller string då? 😅

Reply:
Pga. `formik` + `yup` :p